### ISSUE TITLE
ENHANCEMENT - Publications list configuration

### ISSUE THREAD

**Issue by khawkins98** [visual-framework/vf-wp#240]
Echoes: https://gitlab.ebi.ac.uk/emblorg/backlog/issues/259

**Comment by dbushell** [comment]
We have this config for the (EMBL) Team Publications block plugin (and heading/limit):

<img width="827" alt="team-publications" src="https://user-images.githubusercontent.com/998922/80454386-7cd8eb80-8919-11ea-9dcb-7be37813f21b.png">

is that enough? I will update the [README documentation](https://github.com/visual-framework/vf-wp/blob/master/wp-content/plugins/vf-publications-block/README.md) as it doesn't include the search fields.

The plugin can be configured on deploy same as to how the Jobs and Members plugin are.

**Comment by khawkins98** [comment]
Yeah, that ticket I referenced on GitLab was already sorted (I've now closed it). 

The ask is to create a custom list of publications in the Gutenberg editor; I want to:
-  show N number of publications
-  in a customer order
- filtering publications by:
   - `DOI` or
   - `pubMedId`

That is, don't give me all my publications, but show me publications `C` `Z` and then `A`.
 
Maybe we can do that by:

1. Adding a "filter by DOI or pubMedId"
2. Stacking 3 Gutenberg blocks up

Might also be a need to extend the contentHub here.

Snippet of EDR data:

```json
      {
        "DOI": "10.1074/jbc.RA119.011774",
        "cfIssue": "4",
        "cfNameAbbrev": "J. Biol. Chem.",
        "cfResPublDate": "2020-02-24T00:00:00.000Z",
        "cfStartPage": "1091",
        "cfTitle": "Phosphatidylinositol 3,4-bisphosphate synthesis and turnover are spatially segregated in the endocytic pathway.",
        "cfVol": "295",
        "euroPubMedId": "31831620",
        "originalSource": "MED",
        "pubMedId": "31831620",
        "publYear": "2019",
        "srcAuthors": "Wang H, Loerke D, Bruns C, Müller R, Koch PA, Puchkov D, Schultz C, Haucke V",
        "srcJourName": "The Journal of biological chemistry"
      },
```

**Comment by dbushell** [comment]
Currently blocked by #248

**Comment by khawkins98** [comment]
We should now be able to proceed; #248 is resolved.

**Comment by dbushell** [comment]
Hey @khawkins98 the existing block is using a `limit` query var however the Content Hub no longer seems to respect this. e.g.

https://dev.content.embl.org/api/v1/pattern.html?pattern=embl-team-publications&sort-field-value[changed]=DESC&title=Web%20Development&limit=3

should return only 3 items?

Also for the `DOI` and `pubMedId` are they added as `&pubMedId=31831620` or is there more too it?

**Comment by khawkins98** [comment]
Yeah, this seems like a regression. Should work:

![image](https://user-images.githubusercontent.com/928100/81379552-9ba16400-9109-11ea-9c40-580d475b4f05.png)

https://docs.google.com/document/d/1ywOw18sx3ylP-bO63LEqGGowknuvxYTs5EBwAci1ISU/edit?ts=5bdc201d

@meladawy could you have a look?

**Comment by dbushell** [comment]
I've also notice the Groups theme `template-publications.php` which sets `filter_keyword` and `filter_year` is being ignored by the plugin. Which should allow the front-end user to filter publications on that page.

Note to self: refer to Jobs plugin template for example of using template search/filters.

**Comment by khawkins98** [comment]
Issue now being worked on at https://gitlab.ebi.ac.uk/emblorg/backlog/issues/346

**Comment by khawkins98** [comment]
Updated notes on this after our call on vf-wp today:

## Fixes/updates

These were working but no longer seem to, it may be a "simple" template update needed.

![](https://user-images.githubusercontent.com/998922/80454386-7cd8eb80-8919-11ea-9dcb-7be37813f21b.png)

- [ ] Name of a group or team
   - Group: https://dev.content.embl.org/api/v1/pattern.html?pattern=embl-team-publications&title=Web%20Development
- [ ] ORCID
   - ORCID: https://dev.content.embl.org/api/v1/pattern.html?pattern=embl-person-publications&orcid=0000-0001-5454-2815
- [ ] Name of a person
   - Name: https://dev.content.embl.org/api/v1/pattern.html?pattern=embl-person-publications&title=Maria-Jesus

## New feature

- [ ] Search by DOI of pubMedId
  - We may need contentHub support here ... @meladawy? 

## Blocked

-`limit` and `year` filtering until https://gitlab.ebi.ac.uk/emblorg/backlog/issues/346
